setters/parser.go

```diff
@@ -0,0 +1,107 @@
+package setters
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type (
+	setter interface {
+		set(string, interface{}) error
+	}
+	parser interface {
+		parse(string, bool) (interface{}, error)
+	}
+	parserSetter struct {
+		field  reflect.StructField
+		parser parser
+	}
+)
+
+func (s *parserSetter) set(value string, target interface{}) error {
+	var t = indirectedValue(reflect.ValueOf(target)).FieldByName(s.field.Name)
+
+	v, err := s.parser.parse(value, t.Type().Kind() == reflect.Ptr)
+
+	if err != nil {
+		return err
+	}
+
+	t.Set(reflect.ValueOf(v))
+
+	return nil
+}
+
+type (
+	ErrNotBoolValue struct {
+		value string
+	}
+	ErrSetterNotImplemented struct {
+		field reflect.StructField
+	}
+)
+
+func (e *ErrNotBoolValue) Error() string {
+	return fmt.Sprintf("cfg: Can't parse %q in a bool value", e.value)
+}
+
+func (e *ErrSetterNotImplemented) Error() string {
+	return fmt.Sprintf("cfg: Setter not implemented for type %v", e.field.Type)
+}
+
+type (
+	boolParser   struct{}
```

Comment:
it could be a good idea now to split diferrent parser by domain. `bool_parser.go`, `int_parser.go` etc...